app/models/recipe.rb

```diff
@@ -1,3 +1,11 @@
 class Recipe < ApplicationRecord
-   validates :name, :complexity, :genre, presence: true
+   has_many :ingredients
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   validates :name, :complexity, :genre, presence: true  
+
+   def cost
+      @ingredients.cost.total
+   end
```

Comment:
I do not believe this method is set up properly. Here are a couple of things that could be the issue:
* `@ingredients` does not exist in the context of this class, so it is likely `nil`, and you cannot call methods on `nil`
* If we're using the correct `:ingredients` association, and instead doing `ingredients.cost.total`, `ingredients` returns an AR collection object, i.e. an AR Array, and there is no `cost` method for this AR collection array object.

---

Also, that any methods that you add must also be tested. This method, unfortunately, did not get tested.